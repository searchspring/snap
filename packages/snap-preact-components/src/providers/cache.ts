export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
