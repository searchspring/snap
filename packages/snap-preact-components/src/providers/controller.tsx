import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
