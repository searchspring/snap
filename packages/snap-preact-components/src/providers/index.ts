export * from './store';
export * from './theme';
