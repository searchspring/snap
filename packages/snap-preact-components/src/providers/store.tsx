import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractStore } from '@searchspring/snap-store-mobx';

const StoreContext = createContext<null | AbstractStore>(null);

export const StoreProvider = ({ children, store }: { children: ComponentChildren; store: AbstractStore }) => {
	return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};

export const useStore = () => useContext(StoreContext);

export function withStore<C extends ComponentType>(Component: C): C {
	return ((props: any) => <StoreContext.Consumer>{(store) => <Component {...props} store={store} />}</StoreContext.Consumer>) as C;
}
