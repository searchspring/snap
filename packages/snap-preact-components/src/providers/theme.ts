import { ThemeProvider as EmotionTheme } from '@emotion/react';
export { css, useTheme, withTheme } from '@emotion/react';

interface ThemeProviderProps {
	theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
	children?: any;
}
interface ThemeProvider {
	(props: ThemeProviderProps): React.ReactElement;
}
export const ThemeProvider: ThemeProvider = EmotionTheme;

export const defaultTheme: Theme = {
	colors: {
		primary: '#3A23AD',
		secondary: '#00cee1',
		hover: '#f8f6fd',
		text: {
			secondary: '#ffffff',
		},
		message: {
			error: '#cc1212',
			warning: '#ecaa15',
			info: '#4c3ce2',
		},
	},
	components: {},
};
export interface Theme {
	variables?: {
		[key: string]: unknown;
	};
	colors?: {
		primary: string;
		secondary: string;
		hover?: string;
		text?: {
			secondary: string;
		};
		message?: {
			error: string;
			warning: string;
			info: string;
		};
	};
	components?: any;
	namedComponents?: any;
}
