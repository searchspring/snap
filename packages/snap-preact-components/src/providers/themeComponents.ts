import { BadgeProps } from '../components/Atoms/Badge';
import { BreadcrumbsProps } from '../components/Atoms/Breadcrumbs';
import { ButtonProps } from '../components/Atoms/Button';
import { DropdownProps } from '../components/Atoms/Dropdown';
import { FlexProps } from '../components/Atoms/Flex';
import { FormattedNumberProps } from '../components/Atoms/FormattedNumber';
import { IconProps } from '../components/Atoms/Icon';
import { ImageProps } from '../components/Atoms/Image';
import { LoadingBarProps } from '../components/Atoms/Loading';
import { BannerProps, InlineBannerProps } from '../components/Atoms/Merchandising';
import { OverlayProps } from '../components/Atoms/Overlay';
import { PriceProps } from '../components/Atoms/Price';
import { SkeletonProps } from '../components/Atoms/Skeleton';
import { StringProps } from '../components/Atoms/String';
import { TermsProps } from '../components/Atoms/Terms';
import { ResultLayoutProps } from '../components/Layouts/ResultLayout';
import { CarouselProps } from '../components/Molecules/Carousel';
import { CheckboxProps } from '../components/Molecules/Checkbox';
import { ErrorHandlerProps } from '../components/Molecules/ErrorHandler';
import { FacetGridOptionsProps } from '../components/Molecules/FacetGridOptions';
import { FacetHierarchyOptionsProps } from '../components/Molecules/FacetHierarchyOptions';
import { FacetListOptionsProps } from '../components/Molecules/FacetListOptions';
import { FacetPaletteOptionsProps } from '../components/Molecules/FacetPaletteOptions';
import { FacetSliderProps } from '../components/Molecules/FacetSlider';
import { FilterProps } from '../components/Molecules/Filter';
import { PaginationProps } from '../components/Molecules/Pagination';
import { PerPageProps } from '../components/Molecules/PerPage';
import { RatingProps } from '../components/Molecules/Rating';
import { ResultProps } from '../components/Molecules/Result';
import { SearchInputProps } from '../components/Molecules/SearchInput';
import { SelectProps } from '../components/Molecules/Select';
import { SlideoutProps } from '../components/Molecules/Slideout';
import { SortByProps } from '../components/Molecules/SortBy';
import { BranchOverrideProps } from '../components/Organisms/BranchOverride';
import { FacetProps } from '../components/Organisms/Facet';
import { FacetsProps } from '../components/Organisms/Facets';
import { FilterSummaryProps } from '../components/Organisms/FilterSummary';
import { NoResultsProps } from '../components/Organisms/NoResults';
import { ResultsProps } from '../components/Organisms/Results';
import { SearchHeaderProps } from '../components/Atoms/SearchHeader';
import { SidebarProps } from '../components/Organisms/Sidebar';
import { ToolbarProps } from '../components/Organisms/Toolbar';
import { AutocompleteProps } from '../components/Templates/Autocomplete';
import { RecommendationProps } from '../components/Templates/Recommendation';
import { SearchProps } from '../components/Templates/Search';

type GenericComponentProps<ComponentProps> = Partial<ComponentProps> & { named?: { [named: string]: Partial<ComponentProps> } };

export type ThemeComponents = {
	/* ATOMS */
	badge?: GenericComponentProps<BadgeProps>;
	breadcrumbs?: GenericComponentProps<BreadcrumbsProps>;
	button?: GenericComponentProps<ButtonProps>;
	dropdown?: GenericComponentProps<DropdownProps>;
	flex?: GenericComponentProps<FlexProps>;
	formattedNumber?: GenericComponentProps<FormattedNumberProps>;
	icon?: GenericComponentProps<IconProps>;
	image?: GenericComponentProps<ImageProps>;
	loadingBar?: GenericComponentProps<LoadingBarProps>;
	banner?: GenericComponentProps<BannerProps>;
	inlineBanner?: GenericComponentProps<InlineBannerProps>;
	overlay?: GenericComponentProps<OverlayProps>;
	price?: GenericComponentProps<PriceProps>;
	skeleton?: GenericComponentProps<SkeletonProps>;
	string?: GenericComponentProps<StringProps>;
	terms?: GenericComponentProps<TermsProps>;
	/* MOLECULES */
	carousel?: GenericComponentProps<CarouselProps>;
	checkbox?: GenericComponentProps<CheckboxProps>;
	errorHandler?: GenericComponentProps<ErrorHandlerProps>;
	facetGridOptions?: GenericComponentProps<FacetGridOptionsProps>;
	facetHierarchyOptions?: GenericComponentProps<FacetHierarchyOptionsProps>;
	facetListOptions?: GenericComponentProps<FacetListOptionsProps>;
	facetPaletteOptions?: GenericComponentProps<FacetPaletteOptionsProps>;
	facetSlider?: GenericComponentProps<FacetSliderProps>;
	filter?: GenericComponentProps<FilterProps>;
	pagination?: GenericComponentProps<PaginationProps>;
	perPage?: GenericComponentProps<PerPageProps>;
	rating?: GenericComponentProps<RatingProps>;
	result?: GenericComponentProps<ResultProps>;
	searchInput?: GenericComponentProps<SearchInputProps>;
	select?: GenericComponentProps<SelectProps>;
	slideout?: GenericComponentProps<SlideoutProps>;
	sortBy?: GenericComponentProps<SortByProps>;
	/* ORGANISMS */
	branchOverride?: GenericComponentProps<BranchOverrideProps>;
	facet?: GenericComponentProps<FacetProps>;
	facets?: GenericComponentProps<FacetsProps>;
	filterSummary?: GenericComponentProps<FilterSummaryProps>;
	noResults?: GenericComponentProps<NoResultsProps>;
	results?: GenericComponentProps<ResultsProps>;
	searchHeader?: GenericComponentProps<SearchHeaderProps>;
	sidebar?: GenericComponentProps<SidebarProps>;
	toolbar?: GenericComponentProps<ToolbarProps>;
	/* LAYOUTS */
	resultLayout?: GenericComponentProps<ResultLayoutProps>;
	/* TEMPLATES */
	autocomplete?: GenericComponentProps<AutocompleteProps>;
	recommendation?: GenericComponentProps<RecommendationProps>;
	search?: GenericComponentProps<SearchProps>;
};
