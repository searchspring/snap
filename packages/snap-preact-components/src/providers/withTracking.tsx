import { h, ComponentType, FunctionComponent } from 'preact';
import type { Product } from '@searchspring/snap-store-mobx';
import type { SearchController, AutocompleteController, RecommendationController } from '@searchspring/snap-controller';
import { createImpressionObserver } from '../utilities';
import { useCallback } from 'preact/hooks';

interface WithTrackingProps {
	controller?: SearchController | AutocompleteController | RecommendationController;
	result?: Product;
	[key: string]: any;
}

export function withTracking<Props extends WithTrackingProps>(WrappedComponent: ComponentType<Props>) {
	const WithTracking: FunctionComponent<Props> = (props) => {
		const { controller, result, ...restProps } = props;

		if (!controller) {
			console.warn('Warning: No controller provided to withTracking');
		}

		if (!result) {
			console.warn('Warning: No result provided to withTracking');
		}

		let resetKey;
		if (controller?.type === 'search' || controller?.type === 'autocomplete') {
			const urlManager = (controller as SearchController | AutocompleteController).urlManager;
			resetKey = JSON.stringify({
				q: urlManager.state.query,
				p: urlManager.state.page,
				ps: urlManager.state.pageSize,
				s: urlManager.state.sort,
				f: urlManager.state.filter,
			});
		} else if (controller?.type === 'recommendation') {
			// For recommendations, use a combination of tag and other relevant state
			const recStore = (controller as RecommendationController).store;
			resetKey = JSON.stringify({
				tag: recStore.profile?.tag,
				ids: recStore.results.map((result) => result.id).join(','),
			});
		}

		const { ref, inViewport } = createImpressionObserver({ resetKey });

		if (inViewport) {
			// TODO: add support for disabling tracking events via config like in ResultTracker
			if (result?.type === 'product') {
				controller?.track.product.impression(result as Product);
			} else {
				// track banner in future
			}
		}

		const currentRef = ref.current;
		if (currentRef) {
			const handleClick = useCallback((e: MouseEvent) => {
				controller?.track.product.click(e, result as Product);
			}, []);
			currentRef.setAttribute('sstracking', 'true');
			currentRef.removeEventListener('click', handleClick);
			currentRef.addEventListener('click', handleClick);
		}

		const trackingProps = {
			...restProps,
			controller,
			result,
			trackingRef: ref,
		};

		return <WrappedComponent {...(trackingProps as Props)} />;
	};
	return WithTracking;
}
