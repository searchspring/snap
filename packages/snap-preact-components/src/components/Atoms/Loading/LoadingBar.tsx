/** @jsx jsx */
import { h } from 'preact';

import { jsx, css, keyframes } from '@emotion/react';
import classnames from 'classnames';

import { Theme, useTheme } from '../../../providers/theme';
import { ComponentProps } from '../../../types';

const CSS = {
	loadingBar: ({ color, height, backgroundColor, style, theme, animation }) =>
		css({
			height: height,
			position: 'fixed',
			top: '0',
			left: '0',
			right: '0',
			margin: 'auto',
			transition: 'opacity 0.3s ease',
			opacity: '1',
			visibility: 'visible',
			zIndex: '10000',
			background: backgroundColor || theme.colors?.secondary || '#f8f8f8',

			'& .ss__loading-bar__bar': {
				position: 'absolute',
				top: '0',
				left: '-200px',
				height: '100%',
				background: `${color || theme.colors?.primary || '#ccc'}`,
				animation: `${animation} 2s linear infinite`,
			},
			...style,
		}),
	animation: keyframes({
		from: { left: '-200px', width: '30%' },
		'50%': { width: '30%' },
		'70%': { width: '70%' },
		'80%': { left: '50%' },
		'95%': { left: '120%' },
		to: { left: '100%' },
	}),
};

export function LoadingBar(properties: LoadingBarProps): JSX.Element {
	const globalTheme: Theme = useTheme();
	const theme = { ...globalTheme, ...properties.theme };

	const props: LoadingBarProps = {
		// default props
		height: '5px',
		// global theme
		...globalTheme?.components?.loadingbar,
		// props
		...properties,
		...properties.theme?.components?.loadingbar,
	};

	const { active, color, backgroundColor, height, disableStyles, className, style } = props;

	return (
		active && (
			<div
				css={
					!disableStyles &&
					CSS.loadingBar({
						height,
						color,
						backgroundColor,
						style,
						theme,
						animation: CSS.animation,
					})
				}
				className={classnames('ss__loading-bar', className)}
			>
				<div className="ss__loading-bar__bar"></div>
			</div>
		)
	);
}

export interface LoadingBarProps extends ComponentProps {
	active: boolean;
	color?: string;
	backgroundColor?: string;
	height?: string;
}
