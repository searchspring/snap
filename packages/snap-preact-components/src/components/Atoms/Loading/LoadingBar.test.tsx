import { h } from 'preact';
import { render } from '@testing-library/preact';

import { LoadingBar } from './LoadingBar';
import { ThemeProvider } from '../../../providers';

describe('LoadingBar Component', () => {
	const theme = {
		components: {
			loadingBar: {
				color: 'rgb(200, 100, 50)',
			},
		},
	};

	it('renders', () => {
		const args = {
			active: true,
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar');
		expect(loadingbarElement).toBeInTheDocument();
		expect(loadingbarElement?.classList.length).toBe(2);
	});

	it('displays loading bar when active', () => {
		const args = {
			active: true,
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar');
		expect(loadingbarElement).toBeInTheDocument();
	});

	it('hides loading bar when inactive', () => {
		const args = {
			active: false,
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar');
		expect(loadingbarElement).not.toBeInTheDocument();
	});

	it('uses custom height', () => {
		const args = {
			active: true,
			height: '10px',
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar')!;
		const styles = getComputedStyle(loadingbarElement);
		expect(styles.height).toBe(args.height);
	});

	it('can use custom colors', () => {
		const args = {
			active: true,
			color: 'blue',
			backgroundColor: 'rgb(204, 204, 204)',
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar')!;
		const loadingbarElementStyles = getComputedStyle(loadingbarElement);
		expect(loadingbarElementStyles.background).toBe(args.backgroundColor);

		const barElement = loadingbarElement.querySelector('.ss__loading-bar__bar')!;
		const barElementStyles = getComputedStyle(barElement);
		expect(barElementStyles.background).toBe(args.color);
	});

	it('can disable styling', () => {
		const args = {
			active: true,
			disableStyles: true,
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar');
		expect(loadingbarElement?.classList.length).toBe(1);
	});

	it('can custom className', () => {
		const args = {
			active: true,
			className: 'custom-class',
		};
		const rendered = render(<LoadingBar {...args} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar');
		expect(loadingbarElement?.classList).toContain(args.className);
	});

	it('is themeable with ThemeProvider', () => {
		const args = { active: true };
		const rendered = render(
			<ThemeProvider theme={theme}>
				<LoadingBar {...args} />
			</ThemeProvider>
		);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar .ss__loading-bar__bar')!;
		const styles = getComputedStyle(loadingbarElement);
		expect(styles.backgroundColor).toBe(theme.components.loadingBar.color);
	});

	it('is themeable with theme prop', () => {
		const args = { active: true };
		const rendered = render(<LoadingBar {...args} theme={theme} />);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar .ss__loading-bar__bar')!;
		const styles = getComputedStyle(loadingbarElement);
		expect(styles.backgroundColor).toBe(theme.components.loadingBar.color);
	});

	it('is themeable with theme prop overrides ThemeProvider', () => {
		const args = { active: true };
		const themeOverride = {
			components: {
				loadingBar: {
					color: 'rgb(123, 123, 123)',
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={theme}>
				<LoadingBar {...args} theme={themeOverride} />
			</ThemeProvider>
		);
		const loadingbarElement = rendered.container.querySelector('.ss__loading-bar .ss__loading-bar__bar')!;
		const styles = getComputedStyle(loadingbarElement);
		expect(styles.backgroundColor).toBe(themeOverride.components.loadingBar.color);
	});
});
