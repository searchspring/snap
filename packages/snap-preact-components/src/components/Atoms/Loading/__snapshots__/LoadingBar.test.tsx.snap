// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingBar Component can custom className 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: #ccc;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar custom-class emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component can disable styling 1`] = `
<DocumentFragment>
  <div
    class="ss__loading-bar"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component can use custom colors 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: rgb(204, 204, 204);
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: blue;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component displays loading bar when active 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: #ccc;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component hides loading bar when inactive 1`] = `<DocumentFragment />`;

exports[`LoadingBar Component is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: rgb(200, 100, 50);
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component is themeable with theme prop 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: rgb(200, 100, 50);
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component is themeable with theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: rgb(123, 123, 123);
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component renders 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 5px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: #ccc;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingBar Component uses custom height 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

@keyframes animation-0 {
  from {
    left: -200px;
    width: 30%;
  }

  50% {
    width: 30%;
  }

  70% {
    width: 70%;
  }

  80% {
    left: 50%;
  }

  95% {
    left: 120%;
  }

  to {
    left: 100%;
  }
}

.emotion-0 {
  height: 10px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  margin: auto;
  -webkit-transition: opacity 0.3s ease;
  transition: opacity 0.3s ease;
  opacity: 1;
  visibility: visible;
  z-index: 10000;
  background: #f8f8f8;
}

.emotion-0 .ss__loading-bar__bar {
  position: absolute;
  top: 0;
  left: -200px;
  height: 100%;
  background: #ccc;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div
    class="ss__loading-bar emotion-0"
  >
    <div
      class="ss__loading-bar__bar"
    />
  </div>
</DocumentFragment>
`;
